### PR TITLE
[docs] - Add custom automation conditions guide (DOC-322)

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR adds a customization guide for automation conditions to the docs.

TODOs

- [x] Add screenshots
- [x] Update sidenav
- [x] Add link to Declarative Automation concept page

The busted links in BK will resolve once #23425 is merged.

### CODE REVIEW

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+from dagster import AutomationCondition
+
+any_parents_missing = AutomationCondition.any_deps_match(
+    AutomationCondition.in_progress() | AutomationCondition.failed()
+).with_label("Any parents in progress or failed")
+```
+
+Then, in the UI, the label will display:
+
+<TODO>SCREENSHOT</TODO>
```
Comment (erinkcochran87) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:201]: TODO

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+Hovering over this label will reveal the more complex sub-expressions:
+
+<TODO>SCREENSHOT</TODO>
```
Comment (erinkcochran87) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:205]: TODO

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+)
+```
+
+---
+
+## Examples
+
+### Ignoring specific parents
+
+By default, `AutomationCondition.eager()` will materialize in response to **any** parent update, and not materialize if **any** parent is missing.
```
Comment (erinkcochran87) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:252]: I pulled this straight from the draft in Notion - I think there's something missing here.

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+### Updating in response to code version changes
+
+When you update the an asset's code version, you'll typical want to materialize the asset to reflect the change. To do this, you can use `AutomationCondition.code_version_changed()`:
+
+```jsx
+automation_condition = AutomationCondition.code_version_changed();
+```
+
+This condition will immediately materialize any asset as soon as a code change is detected.
```
Comment (erinkcochran87) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:262]: The Notion draft made it seem like there was supposed to be more here, but we can do a fast-follow here if needed.

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx (hunk 3 above)
Comment (OwenKephart) [reply docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:252]: Ah yeah, the entire Examples section is unfinished at the moment, I think we can cut this all for now

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+Before continuing, you should be familiar with:
+
+- [Asset definitions](/concepts/assets/software-defined-assets)
+- [Declarative Automation](/concepts/automation/declarative-automation)
+
+---
+
+## How it works
+
+Each <PyObject object="AutomationCondition" /> consists of a set of **operands** and various **operators**. To create conditions that suit your specific needs, you can combine the operators and operands listed below with Dagster's built-in <PyObject object="AutomationCondition" pluralize />. For example:
```
Comment (OwenKephart) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:29]: nit: I would consider all of the operators and operands to essentially be "built-in AutomationConditions", so you're really just combining them together in general, maybe ust:

"To create conditions that suit your specific needs, you can combine the operators and operands listed below."

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+```python
+from dagster import AutomationCondition
+
+any_parents_missing = AutomationCondition.any_deps_match(
+    AutomationCondition.in_progress() | AutomationCondition.failed()
+)
+```
+
+This condition translates to **Any upstream dependencies (parents) which are part of an in-progress run or failed during the latest run**.
```
Comment (OwenKephart) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:39]: nit: maybe remove "which" (so just "Any upstream dependencies (parents) are part of ...")

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+any_parents_missing = AutomationCondition.any_deps_match(
+    AutomationCondition.in_progress() | AutomationCondition.failed()
+).with_label("Any parents in progress or failed")
+```
+
+Then, in the UI, the label will display:
+
+<TODO>SCREENSHOT</TODO>
+
+Hovering over this label will reveal the more complex sub-expressions:
```
Comment (OwenKephart) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:203]: (realized that I could take a more informative screenshot): 

"Hovering over this label or expanding the drop-down"

**Review by OwenKephart — APPROVED** [review]
approving modulo small nits and screenshots

File: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+## Describing conditions with labels
+
+When there are a large number of sub-conditions that make up an <PyObject object="AutomationCondition" />, it can be difficult to understand and troubleshoot the condition. To make conditions easier to understand, you can attach labels to sub-conditions, which will then display in the Dagster UI.
+
+Arbitrary string labels can be attached to any node in the <PyObject object="AutomationCondition" /> tree by using the `with_label()` method, allowing you to describe the purpose of a specific sub-condition. For example:
+
+```python
+from dagster import AutomationCondition
+
+any_parents_missing = AutomationCondition.any_deps_match(
```
Comment (OwenKephart) [inline docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx:194]: nit: the name of this variable should be something else (could just be "my_condition")

**Pull request merged**